plugins/Hosting/js/reducers/hosting.js

```diff
@@ -0,0 +1,96 @@
+import { Map, List } from 'immutable'
+import * as constants from '../constants/constants.js'
+
+const initialState = Map({
+	numContracts: 0,
+	storage: 0,
+	usersettings: List([
+		Map({ name: 'Max Duration (Weeks)', value: 0, min: 12 }),
+		Map({ name: 'Collateral per TB per Month (SC)', value: 0 }),
+		Map({ name: 'Price per TB per Month (SC)', value: 0 }),
+		Map({ name: 'Bandwidth Price (SC/TB)', value: 0 }),
+	]),
+	defaultsettings: undefined,
+	files: List([]),
+	earned: 0,
+	expected: 0,
+	acceptingContracts: 0,
+	settingsChanged: false,
+	walletLocked: true,
+	defaultAnnounceAddress: '',
+	modals: Map({
+		shouldShowResizeDialog: false,
+		resizePath: '',
+		resizeSize: 0,
+		initialSize: 0,
+		warningModalTitle: '',
+		warningModalMessage: '',
+		shouldShowWarningModal: false,
+		shouldShowAnnounceDialog: false,
+		announceAddress: '',
+	}),
```

Comment:
@johnathanhowell, thoughts on this pattern? I think your approach has been to split these modals into their own component and pass things like `warningModalTitle` as a prop, rather than making it part of the state.